src/Analyzers/CSharp/CodeFixes/UseParameterNullChecking/CSharpUseParameterNullCheckingCodeFixProvider.cs

```diff
@@ -0,0 +1,109 @@
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseParameterNullChecking
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseIsNullCheck), Shared]
+    internal class CSharpUseParameterNullCheckingCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpUseParameterNullCheckingCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.UseParameterNullCheckingId);
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            context.RegisterCodeFix(
+                new MyCodeAction(CSharpAnalyzersResources.Use_parameter_null_checking,
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var model = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken: cancellationToken);
+                switch (node)
+                {
+                    case IfStatementSyntax ifStatement:
+                        // if (item == null) throw new ArgumentNullException(nameof(item));
+                        var (left, right) = ifStatement.Condition switch
+                        {
+                            BinaryExpressionSyntax binary => (binary.Left, binary.Right),
+                            IsPatternExpressionSyntax isPattern => (isPattern.Expression, ((ConstantPatternSyntax)isPattern.Pattern).Expression),
+                            InvocationExpressionSyntax { ArgumentList.Arguments: var arguments } => (arguments[0].Expression, arguments[1].Expression),
+                            _ => throw ExceptionUtilities.UnexpectedValue(ifStatement.Kind())
+                        };
+
+                        // one of the sides of the binary must be a parameter
+                        var parameterInIf = (IParameterSymbol?)model.GetSymbolInfo(unwrapCast(left), cancellationToken).Symbol
+                            ?? (IParameterSymbol)model.GetSymbolInfo(unwrapCast(right), cancellationToken).Symbol!;
```

Comment:
null suppression worries me.

Reply:
Is unusual for code fixes to make assumptions about what was found by the analyzer?

Reply:
@RikkiGibson I'm not sure, but I thought it is possible that the code changes between the analyzer reporting a diagnostic and the invocation of the <kbd>CTRL</kbd>+<kbd>.</kbd>

Reply:
I would be surprised if the code fix were invoked under conditions where the diagnostic no longer referred to the same thing that was reported by the analyzer. @CyrusNajmabadi @mavasani do you think you could clarify?

Reply:
yes. t his code can make assumptions that everything must be in a correct state as per the analyzer.  *however* a better approach is to have the fixer just include this information in teh 'AdditionalLocations' section of the diagnostic.  Then teh fixer can just pull it out of the diagnostic and not have to compute anything.

Reply:
This code was deleted, so resolving the thread.